src/main/java/com/frostwire/jlibtorrent/CircularArray.java

```diff
@@ -0,0 +1,96 @@
+package com.frostwire.jlibtorrent;
+
+/**
+ * Created on 12/21/16.
+ *
+ * @author aldenml
+ * @author gubatron
+ */
+public class CircularArray {
+    private final int[] buffer;
+    private int head;
+    private int end;
+
+    CircularArray(int capacity) {
+        buffer = new int[capacity];
+        head = -1;
+        end = -1;
+    }
+
+    void add(int value) {
+        if (head == -1) {
+            head = end = 0;
+            buffer[end] = value;
+            return;
+        }
+        end = (end + 1) % buffer.length;
+        buffer[end] = value;
+        if (end <= head) {
+            head = (head + 1) % buffer.length;
+        }
+    }
+
+    int get(int i) {
+// maybe...
+//        if (i > size()-1) {
+//            throw new ArrayIndexOutOfBoundsException();
+//        }
+        return buffer[(head + i) % size()];
```

Comment:
well, the condition is for that, to fail when out of bounds. The reason it currently works (`% size()`) is because since you don't have remove, the `head` ends at most right next to the `end`, but if `remove` is ever implemented, then this will fail.